In [18]:
import numpy as np
import matplotlib.pyplot as plt

In [9]:
import librosa.display
import matplotlib.pyplot as plt


def _plot_signal_and_augmented_signal(signal, augmented_signal, sr):
    fig, ax = plt.subplots(nrows=2)
    librosa.display.waveplot(signal, sr=sr, ax=ax[0])
    ax[0].set(title="Original signal")
    librosa.display.waveplot(augmented_signal, sr=sr, ax=ax[1])
    ax[1].set(title="Augmented signal")
    plt.show()

In [16]:
def add_white_noise(signal, noise_factor):
    noise = np.random.normal(0, signal.std(), signal.size)
    augmented_signal = signal + noise*noise_factor
    return augmented_signal

def time_stretch(signal, stretch_rate):
    return librosa.effects.time_stretch(signal, stretch_rate)

In [21]:
if __name__ == '__main__':
    signal, sr = librosa.load('./data/Asthma/BP1_Asthma,I E W,P L L,70,M.wav')
    print(len(signal), sr)
    

661500 22050


In [34]:
import os

types = os.listdir('./data')
for type in types:
    # print(os.listdir(f'./data/{type}')[-1])
    for audio_file in os.listdir(f'./data/{type}'):
        if audio_file[-3:] != 'wav':
            print(audio_file, type)

desktop.ini Asthma


`desktop.ini` found in Asthma folder.

In [35]:
import os

types = os.listdir('./data')
for type in types:
    # print(os.listdir(f'./data/{type}')[-1])
    for audio_file in os.listdir(f'./data/{type}'):
        if audio_file[-3:] != 'wav':
            os.remove(f'./data/{type}/{audio_file}')

In [38]:
import os

types = os.listdir('./data')
sampling_rate_dictionary = {}

for type in types:
    sampling_rate_dictionary[type] = {}
    # print(os.listdir(f'./data/{type}')[-1])
    for audio_file in os.listdir(f'./data/{type}'):
        signal, sr = librosa.load(f'./data/{type}/{audio_file}')
        if str(sr) not in sampling_rate_dictionary[type]:
            sampling_rate_dictionary[type][str(sr)] = 1
        else:
            sampling_rate_dictionary[type][str(sr)] += 1

In [39]:
for type in types:
    print(sampling_rate_dictionary[type])

{'22050': 99}
{'22050': 105}
{'22050': 15}
{'22050': 63}


As we can see, all sampling rate for all audio files are in 22050 hertz.

We proceed to check the average length of the audio files.

In [59]:
import pandas as pd

def func(x):
    signal, sr = librosa.load(x)
    return len(signal)/sr

for type in types:
    audio_length_series = pd.Series(os.listdir(f'./data/{type}'))
    prefix_series = pd.Series([f'./data/{type}/' for i in range(audio_length_series.size)])
    audio_length_series = prefix_series+audio_length_series
    audio_length_series = audio_length_series.apply(func)
    print(f'{type}')
    print(pd.Series(audio_length_series.value_counts().index.tolist()).describe())
# for type in types:
#     for audio_file in os.listdir(f'./data/{type}'):
#         signal, sr = librosa.load(f'./data/{type}/{audio_file}')
#         if str(sr) not in sampling_rate_dictionary[type]:
#             sampling_rate_dictionary[type][str(sr)] = 1
#         else:
#             sampling_rate_dictionary[type][str(sr)] += 1

Asthma
count    29.000000
mean     16.296847
std       5.408483
min       5.936009
25%      12.528027
50%      15.344036
75%      19.888027
max      30.000000
dtype: float64
Normal
count    34.000000
mean     17.971783
std       4.948170
min       9.840000
25%      13.600023
50%      17.904014
75%      21.584036
max      27.888027
dtype: float64
Pneumonia
count     5.000000
mean     19.580816
std       9.024375
min       9.072018
25%      11.184036
50%      23.472018
75%      24.176009
max      30.000000
dtype: float64
Pulmonary Congestion
count    20.000000
mean     17.049621
std       5.676013
min       8.368027
25%      13.072029
50%      15.248005
75%      22.160023
max      27.568027
dtype: float64
